torch/_functorch/partitioners.py

```diff
@@ -1667,6 +1667,8 @@ def get_recomputable_banned_nodes(banned_nodes: List[fx.Node]) -> List[fx.Node]:
                 # Only allow recomputing nodes that are actually required for BW
                 i.dist_from_bw < int(1e9)  # type: ignore[attr-defined]
                 and get_node_storage(i) not in input_storages
+                and i.meta.get("recompute", CheckpointPolicy.PREFER_SAVE)
+                                != CheckpointPolicy.MUST_SAVE
```

Comment:
ccing @soulitzer in case you have opinions. I think this change seems reasonable. Maybe to restate: the partitioner is generally free to make its own decisions about what to save vs recompute, but it is expected to always respect save/recompute decisions from user `checkpoint` APIs. So in that sense, this PR seems like a strict improvement

Reply:
Agreed, we definitely shouldn't override user decision to save here (esp when none of these APIs are actually peak memory aware)